python/torch_mlir/csrc/backend/LazyShapeInference.cpp

```diff
@@ -0,0 +1,21 @@
+//===- LazyShapeInference.cpp ---------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "LazyShapeInference.h"
+#include "../utils/exception.h"
+
+namespace torch {
+namespace lazy {
+
+std::vector<Shape> compute_shape_detach(const at::Tensor& self) {
```

Comment:
I'd generally encourage you to upstream these to torch to be shared by all backends; I'm not sure if we want to handle the detach op in other backends though